### PR TITLE
julia 0.4 --> 0.5

### PR DESCRIPTION
**Pull request by jeff-regier**
Hi @kbarbary -- I made a fork of WCS.jl that gets rid of the latest deprecated warning from Julia 0.5. It isn't backward compatible with 0.4 but I thought I'd submit the PR in case you wanted to merge it anyway.  (Presumably users still on 0.4 would fetch the latest tagged version that's compatible with their version of Julia if they do `Pkg.add("WCS")`. )

### CODE REVIEW

**Comment by giordano** [comment]
Why not using [`Compat.jl`](https://github.com/JuliaLang/Compat.jl) instead of dropping support for Julia 0.4 that is still the current stable version of Julia?

**Comment by jeff-regier** [comment]
I tried with Compat, but I wasn't able to make one of the `Array{String}`
variables compatible with an `Array{ByteString}`.

**Comment by kbarbary** [comment]
Thanks Jeff! I'm sure there is a way to make it work on both versions, perhaps using raw `if VERSION >= 0.5.0-dev+???? ... else ... end`.

That said, I'm fine with just dropping support for v0.4 in the dev version of WCS. It's not like we're rapidly adding new features to WCS that should be backported to support v0.4. The only downside is that if we do this, we should backport any bugfixes to a WCS v0.1.x release branch for a while.

**Comment by tkelman** [comment]
You can use Compat.ASCIIString. Why is this package defining Base.convert between Base types? That means you change the behavior of other code depending on whether or not this package is loaded, which is really poor form to do. You can extend Base methods on types defined in the package, but extending Base methods on Base types shouldn't be done by packages.

**Comment by kbarbary** [comment]
@tkelman I hadn't thought of that repercussion. My simple thinking was that I wanted functionality that falls under the description of `Base.convert`, so it seemed like it made sense to add a method to that function. Didn't think about the difference between Base types and types in the module.

Do you know if this point is emphasized in the Julia docs somewhere?

**Comment by jeff-regier** [comment]
Has been any discussion of changing Julia, to disallow overriding methods
whose argument types are all defined in other modules? (Are there any
situations where that behavior would be useful?)

**Comment by tkelman** [comment]
for nefarious purposes, or wanting to change things at the REPL perhaps. I'm not sure if the base language can warn for this, but it should go into a linter if Lint.jl doesn't already check for it.

File: REQUIRE [hunk 1]
```diff
@@ -1,2 +1,3 @@
-julia 0.5-dev
+julia 0.4+
```
Comment (tkelman) [inline REQUIRE]: the `+` is not usually what you want here http://docs.julialang.org/en/release-0.4/manual/strings/#version-number-literals

File: REQUIRE [hunk 2]
```diff
@@ -1,2 +1,3 @@
...
 BinDeps
+Compat
```
Comment (tkelman) [inline REQUIRE]: best to specify a minimum version number that implements the features you're using

**Comment by jeff-regier** [comment]
Do either of you see why the tests don't pass on a mac? It's something about the changes to build.jl. I don't have a mac though so it's hard to figure out.

The error on travis is
```
LoadError: File AbstractString["/Users/travis/.julia/v0.4/WCS/deps/usr/lib/libwcs.so.5.13"] was not created successfully (Tried to run BinDeps.SynchronousStepCollection(Any[`./configure --prefix=/Users/travis/.julia/v0.4/WCS/deps/usr --disable-fortran --without-cfitsio --without-pgplot --disable-utils`,`make install`,`ln -s /Users/travis/.julia/v0.4/WCS/deps/usr/lib/libwcs.5.dylib /Users/travis/.julia/v0.4/WCS/deps/usr/lib/libwcs.dylib`],"","") )
while loading /Users/travis/.julia/v0.4/WCS/deps/build.jl, in expression starting on line 39
```

File: deps/build.jl [hunk 3]
```diff
@@ -11,8 +13,12 @@ builddir = joinpath(depsdir, "src/wcslib-$version")
 prefix = joinpath(depsdir, "usr")
 configopts = ["--disable-fortran", "--without-cfitsio", "--without-pgplot",
               "--disable-utils"]
-@unix_only libfilename = "libwcs.so.$version"
-@osx_only libfilename = "libwcs.$version.dylib"
+if is_unix()
+    libfilename = "libwcs.so.$version"
+elseif is_apple()
+    libfilename = "libwcs.$version.dylib"
+end
```
Comment (kbarbary) [inline deps/build.jl]: `is_unix()` evaluates true for both Linux and Mac. I think you want `is_linux()`.

File: deps/build.jl [hunk 4]
```diff
@@ -13,7 +13,7 @@ builddir = joinpath(depsdir, "src/wcslib-$version")
 prefix = joinpath(depsdir, "usr")
 configopts = ["--disable-fortran", "--without-cfitsio", "--without-pgplot",
               "--disable-utils"]
-if is_unix()
+if is_linux()
```
Comment (tkelman) [inline deps/build.jl]: switch the order. there are unixes other than apple and linux.

**Comment by jeff-regier** [comment]
Passing, at last!

**Comment by jeff-regier** [comment]
@kbarbary  -- When you have a chance, would you tag a new version for this PR? I'd like `Pkg.resolve("Celeste")` to find the a version of `WCS.jl` that works well with Julia 0.5. Thanks!

**Comment by kbarbary** [comment]
LGTM, merging. I'll fix the overloading of `Base.convert` that Tony brought up in a separate PR.

**Pull request merged**